src/output/plugins/visualization/LowLevelProtocol.hxx

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The Music Player Daemon Project
+
+#ifndef LOW_LEVEL_PROTOCOL_HXX_INCLUDED
+#define LOW_LEVEL_PROTOCOL_HXX_INCLUDED
+
+#include <fftw3.h>
+
+#include <algorithm>
+#include <cstdint>
+
+#include <netinet/in.h>
```

Comment:
That header isn't portable, is it?
I suggest using MPD's `util/ByteOrder.hxx` header, which is portable and even faster!
It also has a `PackedBE32` class which you can use to write big-endian 32 bit integers to a byte stream (without reverting to `std::copy()`).

Reply:
Huh... I had no idea (I only work on Linux). Replaced... but I don't see how to use `PackedBE{16,32}` to copy the raw bytes into a buffer without recourse to `std::copy` (or its equivalent)?

Reply:
Since the `Packed` classes are just bytes, you can simply cast a pointer to `PackedBE16*` and access the object from there.